### PR TITLE
Budget app

### PR DESCRIPTION
**Pull request by karam084**
## Design

- You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:

## Colors.

- Typography: font face, size and weight.
- Layout: composition and space between elements.
- NOTE: In these design guidelines there are several UIs that you won't need for this exercise; also, some pages are not given a design and you will create them following the design guidelines of the other pages.
- Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).
- The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

## Interactions
- Splash screen

- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in page
- The user should be able to register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (catagories page)
- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

### Transactions page

- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page
- The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,4 +131,111 @@ Things you may want to cover:
...
+
+- Hat tip to anyone whose code was used
+- Inspiration
+- etc
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+# Demo video
+[loom video](https://www.loom.com/share/4d3d208a3717479f9db0696084c11653)
+
```
Comment (mahbubul14) [inline README.md]: - [x] The video should contain a highlight about a piece of the code you find interesting or proud of making. :+1:

File: README.md [hunk 2]
```diff
@@ -1,9 +1,119 @@
-# README
+<a name="budget-app"></a>
```
Comment (mahbubul14) [inline README.md:2]: - [x] You should deploy the application using Heroku, Netlify or GitHub Pages. Kindly work on it. :+1:

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @karam084,

Good job so far! :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "fix bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: README.md (hunk 1 above)
Comment (karam084) [reply README.md]: Solved thank you

File: README.md (hunk 2 above)
Comment (karam084) [reply README.md:2]: Solved thank you

File: README.md [hunk 3]
```diff
@@ -1,9 +1,119 @@
...
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+ # Live Demo
 
-Things you may want to cover:
+ [link](https://budget-app-ohsx.onrender.com)
```
Comment (mikethreels) [inline README.md:5]: - [ ] your project should be deployed using Heroku, Netlify or Github pages
- [ ] when trying to sign in or sign up to your live demo application I get the below error, please try to fix this. 
afbeelding

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Karam, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- linters are passing

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 3 above)
Comment (karam084) [reply README.md:5]: Thank you for reviewing I have an error with Heroku required to add a payment method Netlify does not work with rail and the same git-hub not working.
Thank you for understanding my case.

File: README.md [hunk 4]
```diff
@@ -21,4 +131,111 @@ Things you may want to cover:
...
+<!-- LICENSE -->
+
+## Design Template
+
+-  I give credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the author of the [original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)
+
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (rloterh) [inline README.md:239]: - [x] OPTIONAL
MIT license link in your ReadMe is broken or is not set, kindly endeavor to fix it.

File: README.md [hunk 5]
```diff
@@ -21,4 +131,111 @@ Things you may want to cover:
...
+To run tests, please run this command
+
+$ rspec ./spec/#{filename}_spec.rb 
+To run all tests, please run this command
+
+or $ spec  
+
+<!-- AUTHORS -->
+
+## 👥 Authors <a name="authors"></a>
```
Comment (rloterh) [inline README.md:179]: - [x] OPOTIONAL
I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by a sole contributor.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Fouad,  

Your project is complete! Great job! There is nothing else to say other than... it's time to merge it 🍾🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Properly added required models, controllers, and corresponding views
- Devise correctly implemented for authentication
- Schema follow given ERD
- App is running as intended

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

**You can also consider:**
- You used incorrect gitflow/ GitHub flow for this project! Kindly note that capstone project worked on by a single contributor should have merge request flow **from feature into development**. Importantly, kindly ensure to use the correct gitflow for subsequent capstone projects.

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 5 above)
Comment (karam084) [reply README.md:179]: Solved, Thank you

File: README.md (hunk 4 above)
Comment (karam084) [reply README.md:239]: Solved, Thank you

**Comment by karam084** [comment]
Fix linter

**Pull request merged**